core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -249,32 +256,36 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
         requireCreateWorkspaceAction(billingProject.projectName, childContext)
       }
 
+      // get the billing _profile_ from BPM; this holds the cloud billing account information related to the billing
+      // project (like the cloud platform and the billing account id.) A billing _profile_ is not guaranteed to be
+      // present for a given billing project, but if it is, we'll use it to determine whether to create a multi-cloud
+      // workspace or a rawls workspace
       billingProfileOpt <- traceFutureWithParent("getBillingProfile", parentContext) { s =>
         getBillingProfile(billingProject, s)
       }
 
-      workspaceOpt <- Apply[Option]
-        .product(Option(multiCloudWorkspaceConfig.azureConfig), billingProfileOpt)
-        .traverse { case (azureConfig, profileModel) =>
-          // "MultiCloud" workspaces are limited to azure-hosted workspaces for now.
-          // This will likely change when the functionality for GCP workspaces gets moved out of Rawls
-          Option(profileModel.getCloudPlatform)
-            .filter(_ == CloudPlatform.AZURE)
-            .traverse { _ =>
-              traceFutureWithParent("createMultiCloudWorkspace", parentContext) { s =>
-                createMultiCloudWorkspace(
-                  workspaceRequest,
-                  profileModel,
-                  s
-                ).map(workspace => (workspace, WorkspaceCloudPlatform.Azure))
-              }
+      workspaceOpt <-
+        if (isMcRequest(workspaceRequest, billingProfileOpt)) {
+          billingProfileOpt
+            .traverse { profileModel =>
```

Comment:
I haven't compiled but -- does this give you an `Option[Option[..]]`? This may be a use case for `.flatTraverse`

Reply:
Check out https://github.com/broadinstitute/rawls/pull/2981/commits/5a104207b93b51dfbf2d97dd95e094c954f29f1b